modules/rostests/win32/user32/serverside/serverside.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * PROJECT:         ReactOS test suite
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Test for existense of the kernel mode wndproc in the window
+ * COPYRIGHT:       Copyright 2024 Oleg Dubinskiy (oleg.dubinskiy@reactos.org)
+ */
+
+/* This testapp tests behaviour of IsServerSideWindow() function in user32. */
+
+#define _WINE
+
+#include <windows.h>
+#include <stdio.h>
+
+#include <debug.h>
+
+WCHAR WndClass[] = L"window class";
+
+LRESULT CALLBACK WndProc(HWND hWnd,
+                         UINT msg,
+                         WPARAM wParam,
+                         LPARAM lParam)
+{
+    switch (msg)
+    {
+        case WM_PAINT:
+            return 0;
+
+        case WM_DESTROY:
+            PostQuitMessage(0);
+            return 0;
+    }
+
+    return DefWindowProcW(hWnd, msg, wParam, lParam);
+}
+
+int APIENTRY wWinMain(HINSTANCE hInst,
+                      HINSTANCE hPrevInstance,
+                      LPWSTR lpCmdLine,
+                      int nCmdShow)
+{
+    HWND hWnd1a, hWnd1b;
+    WNDCLASSEXW wcx;
+    UINT result;
+    BOOL ret;
+
+    memset(&wcx, 0, sizeof(wcx));
+    wcx.cbSize = sizeof(wcx);
+    wcx.lpfnWndProc = (WNDPROC)WndProc;
+    wcx.hInstance = hInst;
+    wcx.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1);
+    wcx.lpszClassName = WndClass;
+
+    if (!(result = RegisterClassExW(&wcx)))
+        return 1;
+
+    /* 1. Window with a valid wndproc */
+    hWnd1a = CreateWindowExW(0,
+                             WndClass,
+                             NULL,
+                             WS_CAPTION | WS_SYSMENU,
+                             CW_USEDEFAULT, CW_USEDEFAULT,
+                             400, 100,
+                             NULL, 0,
+                             hInst, NULL);
+    if (!hWnd1a)
+        return 1;
+
+    ShowWindow(hWnd1a, SW_SHOW);
+    UpdateWindow(hWnd1a);
+
+    ret = IsServerSideWindow(hWnd1a);
```

Comment:
It would be of interest to check whether the last-error has been set to something, or is left unchanged.